src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -0,0 +1,99 @@
+namespace CarbonAware.WepApi.IntegrationTests;
+
+using CarbonAware.Tools.WattTimeClient;
+using Microsoft.AspNetCore.Mvc.Testing;
+using NUnit.Framework;
+using System.Net;
+using WireMock.Server;
+
+public class APIWebApplicationFactory : WebApplicationFactory<Program> {
+
+}
+
+/// <summary>
+/// Tests that the Web API controller handles and packages various responses from a plugin properly 
+/// including empty responses and exceptions.
+/// </summary>
+[TestFixture]
+public class CarbonAwareControllerTests
+{
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+	private APIWebApplicationFactory _factory;
+	private HttpClient _client;
+    private WireMockServer _server;
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    private bool resetEnvironment = false;
+
+    [OneTimeSetUp]
+    public void Setup()
+    {
+		//Defaults to JSON Integration Testing
+		//WattTime can be tested by uncommenting the line below or setting an env variable
+		//Environment.SetEnvironmentVariable("CarbonAwareVars__CarbonIntensityDataSource", "WattTime");
+
+		if (Environment.GetEnvironmentVariable("CarbonAwareVars__CarbonIntensityDataSource") == "WattTime")
```

Comment:
Did we intend for these unit tests to ONLY run when a pre-existing environment variable is set?  Should we be setting the environment variable to WattTime any time these tests are run?

Reply:
This is actually acting as a switch between either JSON or WattTime testing. If the environment variable is unset, or set to "json", the tests do not spin up a wiremock server and test the JSON path instead. This allows for the integration tests to be run multiple times with different environment variables set in pipeline.